pbparam/cost_functions/base_cost_function.py

```diff
@@ -0,0 +1,17 @@
+#
+# Base cost function class
+#
+
+class BaseCostFunction():
+    """
+    Base cost function class
+    """
+    def __init__(self):
+        # super().__init__()
+        # self.name = "Cost function"
+        self.y_sim = None
+        self.y_data = None
+        self.sd = None
```

Comment:
You don't need this as the data and parameters will be passed at every call of the cost_function.